Source/Diagnostics/Diagnostics.cpp

```diff
@@ -199,29 +200,34 @@ Diagnostics::InitData ()
             InitializeFieldBufferData(i_buffer, lev);
         }
     }
-    // When particle buffers, m_particle_boundary_buffer are included,
-    // they will be initialized here
-    InitializeParticleBuffer();
 
     amrex::ParmParse pp_diag_name(m_diag_name);
+    // default for writing species output is 1
+    int write_species = 1;
+    pp_diag_name.query("write_species", write_species);
+    if (write_species == 1) {
+        // When particle buffers, m_particle_boundary_buffer are included,
+        // they will be initialized here
+        InitializeParticleBuffer();
+        InitializeParticleFunctors();
+    }
+
     amrex::Vector <amrex::Real> dummy_val(AMREX_SPACEDIM);
     if ( queryArrWithParser(pp_diag_name, "diag_lo", dummy_val, 0, AMREX_SPACEDIM) ||
          queryArrWithParser(pp_diag_name, "diag_hi", dummy_val, 0, AMREX_SPACEDIM) ) {
         // set geometry filter for particle-diags to true when the diagnostic domain-extent
         // is specified by the user
         for (int i = 0; i < m_output_species.size(); ++i) {
-            m_output_species[i].m_do_geom_filter = true;
+            m_output_species[0][i].m_do_geom_filter = true;
```

Comment:
Maybe we need a comment here and 5 lines below what is in `[0]` again.
Cannot quickly understand it without going back to definition and seeing how it is used elsewhere.